Revise onboarding document with project details

amponce opened the pull request:
Removed Quick Reference Checklist section and added project overview and team contacts.

amponce committed:
Revise onboarding document with project details

Removed Quick Reference Checklist section and added project overview and team contacts.

michaelclement commented on teams/benefits-portfolio/benefits-intake-optimization/engineering/Onboarding.md:
SOCKS was killed off so I think we can remove that first point

```suggestion
1. **GitHub access** to department-of-veterans-affairs repositories
2. **Staging environment credentials** (test user accounts like vets.gov.user+228@gmail.com)
3. **Datadog access** for monitoring and debugging
4. **Slack access** to #benefits-intake-optimization and platform channels
```

amponce committed:
Update teams/benefits-portfolio/benefits-intake-optimization/engineering/Onboarding.md

Co-authored-by: Michael Clement <18408628+michaelclement@users.noreply.github.com>

amponce committed:
Update teams/benefits-portfolio/benefits-intake-optimization/engineering/Onboarding.md

Co-authored-by: Michael Clement <18408628+michaelclement@users.noreply.github.com>

michaelclement commented on teams/benefits-portfolio/benefits-intake-optimization/engineering/Onboarding.md:
We can link to the mocking details in the vets-website readme [here](https://github.com/department-of-veterans-affairs/vets-website?tab=readme-ov-file#running-a-mock-api-for-local-development):

```suggestion
4. Configure [mock API](https://github.com/department-of-veterans-affairs/vets-website?tab=readme-ov-file#running-a-mock-api-for-local-development) or connect to staging backend
```

amponce committed:
Update yarn install command in onboarding guide

amponce committed:
Remove Staging Review Process from Onboarding.md

Removed the Staging Review Process section and related artifacts from the Onboarding document.

amponce commented on teams/benefits-portfolio/benefits-intake-optimization/engineering/Onboarding.md:
✅
